.ci/releasePipeline.groovy

```diff
@@ -141,3 +127,175 @@ pipeline {
         }
     }
 }
+
+
+String getVersionFromGitTagName() {
+    def ver = sh(returnStdout: true, script: "git tag -l --points-at HEAD --sort=-creatordate | head -n 1")
+        .trim()
+        .substring(1)  // drop v prefix
+    echo "version= ${ver}"
+    return ver
+}
+
+void getSignClient() {
+    withVault(vaultSecrets: [[path        : 'keptn-jenkins/monaco/sign-service',
+                              secretValues: [[envVar: 'repository', vaultKey: 'client_repository_url ', isRequired: true]]]]
+    ) {
+        if (!fileExists('sign-client.jar')) {
+            sh '''
+                curl --request GET $repository/testautomation-release-local/com/dynatrace/services/signservice/client/1.0.[RELEASE]/client-1.0.[RELEASE].jar
+                     --show-error
+                     --silent
+                     --globoff
+                     --fail-with-body
+                     --output sign-client.jar
+               '''.replaceAll("\n", " ").replaceAll(/ +/, " ")
+        }
+    }
+}
+
+void buildBinary(Map args = [command: null, version: null, dest: null]) {
+    stage('🏁 Build ' + args.command) {
+        sh "make ${args.command} VERSION=${args.version} OUTPUT=${args.dest}"
+    }
+}
+
+void signWinBinaries(Map args = [source: null, version: null, destDir: null, projectName: null]) {
+    stage('Sign binaries') {
+        signToSignService(source: args.source, version: args.version, destDir: args.destDir, signAction: "SIGN", projectName: args.projectName)
+    }
+}
+
+void createShaSum(Map args = [source: null, dest: null]) {
+    stage('Compute SHA sum') {
+        sh "sha256sum ${args.source} | sed 's, .*/, ,' > ${args.dest}"
```

Comment:
what exactly does that sed expression replace? I assume some line ending characters that caused issues with sign service when I tested - and why not just have shasum omit the line break (-z) as my version had

Did you test the verification steps locally?

Reply:
in a path of file for which are sha sum computed, this removes leading part of path if `sha256sum` command isn't executed in same directory

It is checked once, but will do it again when token will be ready, and the whole procedure will pass from the beginning to the end